unit_tests/multiphase/test_vof_cons.cpp

```diff
@@ -0,0 +1,237 @@
+
+        // Set timestep according to input
+        dt = ft_time / ((amrex::Real)nx) * CFL;
+        // Round to nearest integer timesteps
+        int niter = (int)round(ft_time / dt);
+        // Modify dt to fit niter
+        dt = ft_time / ((amrex::Real)niter);
+
+        populate_parameters();
+        {
+            amrex::ParmParse pp("geometry");
+            amrex::Vector<int> periodic{{1, 1, 1}};
+            pp.addarr("is_periodic", periodic);
+        }
+
+        initialize_mesh();
+
+        auto& repo = sim().repo();
+
+        // PDE manager, for access to VOF PDE later
+        auto& pde_mgr = sim().pde_manager();
+        // Setup of icns provides MAC velocities
+        pde_mgr.register_icns();
+
+        // Initialize physics for the sake of MultiPhase routines
+        sim().init_physics();
+
+        // Initialize volume fraction field
+        auto& vof = repo.get_field("vof");
+        run_algorithm(vof, [&](const int lev, const amrex::MFIter& mfi) {
+            auto vof_arr = vof(lev).array(mfi);
+            const auto& bx = mfi.validbox();
+            initialize_volume_fractions(dir, bx, vof_arr);
+        });
+        // Populate boundary cells
+        vof.fillpatch(0.0);
+        // Get multiphase object
+        auto& mphase = sim().physics_manager().get<amr_wind::MultiPhase>();
+
+        // Initialize constant velocity field in single direction
+        amrex::Array<amrex::Real, 3> varr = {0};
+        if (dir < 0) {
+            varr[0] = m_vel;
+            varr[1] = m_vel;
+            varr[2] = m_vel;
+        } else {
+            varr[dir] = m_vel;
+        }
+
+        // Set advection velocities accordingly
+        auto& umac = repo.get_field("u_mac");
+        auto& vmac = repo.get_field("v_mac");
+        auto& wmac = repo.get_field("w_mac");
+        run_algorithm(vof, [&](const int lev, const amrex::MFIter& mfi) {
+            auto um = umac(lev).array(mfi);
+            auto vm = vmac(lev).array(mfi);
+            auto wm = wmac(lev).array(mfi);
+            const auto& gbx = mfi.growntilebox(1);
+            amrex::ParallelFor(
+                gbx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                    um(i, j, k) = varr[0];
+                    vm(i, j, k) = varr[1];
+                    wm(i, j, k) = varr[2];
+                });
+        });
+
+        // Get initial volume fraction
+        amrex::Real sum_vof0 = mphase.volume_fraction_sum();
+
+        for (int n = 0; n < niter; ++n) {
+            // Perform VOF solve
+            for (auto& seqn : pde_mgr.scalar_eqns()) {
+                if (seqn->fields().field.base_name() == "vof") {
+                    seqn->initialize();
+                    seqn->compute_advection_term(amr_wind::FieldState::Old);
+                    seqn->post_solve_actions();
+                }
+                // Check conservation
+                EXPECT_NEAR(mphase.volume_fraction_sum(), sum_vof0, tol);
+            }
+        }
```

Comment:
You don't need the loop in L173. You should be able to access the VOF equation directly from the `pde_mgr` object  before the for loop in L171. 

```C++
auto seqn = pde_mgr<amr_wind::pde::VOF>(amr_wind::pde::VOF::pde_name());
```

Why is the `seqn->initialize()` called within each iteration?